packages/miniflare/src/workers/core/entry.worker.ts

```diff
@@ -351,6 +355,156 @@ async function handleScheduled(
+	const clonedRequest = request.clone();
+	if (!request.body || !clonedRequest.body) {
+		return new Response("Include an email in the body", { status: 400 });
+	}
+
+	// Get request blob as we need its size
+	const incomingEmailRaw = await (await request.blob()).bytes();
+
+	// Email worker limit is 25 MiB total
+	if (incomingEmailRaw.byteLength > 25 * 1024 * 1024) {
+		return new Response("Email is bigger than current limit - 25MiB.", {
+			status: 400,
+		});
+	}
+
+	let parsedIncomingEmail: Email;
+	try {
+		parsedIncomingEmail = await PostalMime.parse(incomingEmailRaw);
+	} catch (e) {
+		const error = e as Error;
+		return new Response(
+			`Email could not be parsed: ${error.name}: ${error.message}`,
+			{ status: 400 }
+		);
+	}
+
+	if (parsedIncomingEmail.messageId === undefined) {
+		return new Response(
+			"Email could not be parsed: invalid or no message id provided",
+			{ status: 400 }
+		);
+	}
+
+	if (from !== parsedIncomingEmail.from.address) {
+		console.log(
+			`${yellow("Provided envelope from address doesn't match message From header")} - Envelope From: ${from}; Header From: ${parsedIncomingEmail.from.address}`
+		);
+	}
+
+	if (!parsedIncomingEmail.to?.map((addr) => addr.address).includes(to)) {
+		console.log(
+			`${yellow("Provided envelope to address doesn't match any message To address")} - Envelope To: ${to}; Header To: ${parsedIncomingEmail.to?.map((addr) => addr.address)}`
+		);
+	}
+
+	let incomingEmailHeaders: Headers;
+	try {
+		incomingEmailHeaders = new Headers(
+			parsedIncomingEmail.headers.map((header) => [header.key, header.value])
+		);
+	} catch (e) {
+		const error = e as Error;
+		return new Response(
+			`Email could not be parsed: ${error.name}: ${error.message}`,
+			{ status: 400 }
+		);
+	}
+
+	let maybeClientError: string | undefined = undefined;
+
+	// @ts-expect-error .email is not in the fetcher but it's a valid RPC call.
+	await service.email(
+		// FIXME(lduarte): I have a feeling that because we are using RPC to do this, from + to + headers + readableStream can only take
+		// up 1MiB. Headers will probably be the part that takes up most space, but it's rare for emails headers to be that big.
+		createForwardableEmailMessage(
+			from,
+			to,
+			clonedRequest.body,
+			incomingEmailHeaders,
+			function setReject(reason: string): void {
+				console.log(
+					`${red(".setReject() called from Email Handler")} with the following reason: "${reason}"`
+				);
+				maybeClientError = reason;
+			},
+			async function forward(rcptTo: string, headers?: Headers): Promise<void> {
+				console.log(
+					`${blue(".forward() called from Email Handler")} with\n  rcptTo: ${rcptTo}${headers ? `\n  headers:\n${[...headers.entries()].map(([k, v]) => `    ${k}: ${v}`).join("\n")}` : ""}`
+				);
+			},
+			async function reply(replyMessage: EmailMessage): Promise<void> {
+				if (!isEmailReplyable(parsedIncomingEmail, incomingEmailHeaders)) {
+					throw new Error("original email is not repliable");
```

Comment:
```suggestion
					throw new Error("Original email is not replyable");
```

Will users know what this means? What the action here for them to fix this error?

Reply:
i tried to have error messages as closely matched as production, I admit that they aren't the best ones, but maybe it's more worth to have an environment close to prod?

Reply:
Should we not fix production as well, if that typo is there?